### PR TITLE
pybind/mgr/pg_autoscale: revert to default profile scale-up

### PR DESCRIPTION
**Pull request by kamoltat**
pg_autoscale module will now start out all the pools
with a scale-up profile by default.

Added tests in workunits/mon/pg_autoscaler.sh
to evaluate if the default pool creation is
a scale-up profile

Updated documentation and release notes to
reflect the change in the default behavior
of the pg_autoscale profile.

Fixes: https://tracker.ceph.com/issues/53309

Signed-off-by: Kamoltat <ksirivad@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests
- Teuthology
  - [ ] Completed teuthology run
  - [ ] No teuthology test necessary (e.g., documentation)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by kamoltat** [comment]
https://pulpito.ceph.com/ksirivad-2021-11-18_19:50:17-rados:singleton:all:pg-autoscaler.yaml-wip-autoscale-profile-scale-up-default-distro-basic-smithi/6512431/

**Comment by kamoltat** [comment]
Test:
https://pulpito.ceph.com/ksirivad-2021-11-18_21:54:49-rados:singleton:all:pg-autoscaler.yaml-wip-autoscale-profile-scale-up-default-distro-basic-smithi/

**Comment by neha-ojha** [comment]
> Test: https://pulpito.ceph.com/ksirivad-2021-11-18_21:54:49-rados:singleton:all:pg-autoscaler.yaml-wip-autoscale-profile-scale-up-default-distro-basic-smithi/

This is still failing with

```
2021-11-18T22:19:41.086 INFO:tasks.workunit.client.0.smithi058.stderr:+ echo 'Error: a PROFILE is scale-down'
2021-11-18T22:19:41.087 INFO:tasks.workunit.client.0.smithi058.stdout:Error: a PROFILE is scale-down
2021-11-18T22:19:41.087 DEBUG:teuthology.orchestra.run:got remote process result: 1
```

**Comment by kamoltat** [comment]
1/1 passed
https://pulpito.ceph.com/ksirivad-2021-11-19_15:13:16-rados:singleton:all:pg-autoscaler.yaml-wip-autoscale-profile-scale-up-default-distro-basic-smithi/

File: qa/workunits/mon/pg_autoscaler.sh [hunk 1]
```diff
@@ -45,6 +45,35 @@ ceph osd pool set b pg_autoscale_mode on
...
+  echo "Error: b PROFILE is scale-down"
+  exit 1
+fi
+
+# This part of this code will now evaluate the accuracy of
+# scale-down profile
+
+# change to scale-down profile
+
+ceph osd pool set autoscale-profile scale-down
```
Comment (neha-ojha) [inline qa/workunits/mon/pg_autoscaler.sh:75]: In https://pulpito.ceph.com/ksirivad-2021-11-19_15:13:16-rados:singleton:all:pg-autoscaler.yaml-wip-autoscale-profile-scale-up-default-distro-basic-smithi/, why do we see "already a scale-down"? Shouldn't we do a similar check for autoscale-status here like above?

```
2021-11-19T15:30:30.855 INFO:tasks.workunit.client.0.smithi116.stdout:Success: pool b PROFILE is scale-up
2021-11-19T15:30:30.855 INFO:tasks.workunit.client.0.smithi116.stderr:+ echo 'Success: pool b PROFILE is scale-up'
2021-11-19T15:30:30.856 INFO:tasks.workunit.client.0.smithi116.stderr:+ ceph osd pool set autoscale-profile scale-down
2021-11-19T15:30:31.221 INFO:tasks.workunit.client.0.smithi116.stderr:autoscale-profile is already a scale-down!
```

**Comment by neha-ojha** [comment]
jenkins test make check

**Review by neha-ojha — APPROVED** [review]
Let's also run this through a broader rados suite

**Comment by neha-ojha** [comment]
Being tested at: https://pulpito.ceph.com/ksirivad-2021-11-19_19:14:07-rados-wip-autoscale-profile-scale-up-default-distro-basic-smithi/

**Comment by neha-ojha** [comment]
> Being tested at: https://pulpito.ceph.com/ksirivad-2021-11-19_19:14:07-rados-wip-autoscale-profile-scale-up-default-distro-basic-smithi/

no relates failures, unrelated failures tracked in https://pad.ceph.com/p/rado_suite_test_laura_junior

**Pull request merged**